aosp_diff/preliminary/bionic/0006-bionic-libm-avx2-optimization.patch

```diff
@@ -0,0 +1,1045 @@
++            shared: {
++               srcs: ["x86_64/dynamic_function_dispatch.cpp"],
++            },
++            static: {
++               srcs: ["x86_64/static_function_dispatch.S"],
++            },
+             version_script: ":libm.x86_64.map",
+         },
+     },
+@@ -587,3 +611,70 @@ genrule {
+     tool_files: [":bionic-generate-version-script"],
+     cmd: "$(location :bionic-generate-version-script) x86_64 $(in) $(out)",
+ }
++
++cc_library_static {
++    name: "msun_fma_avx2_math",
++    ramdisk_available: true,
++    vendor_ramdisk_available: true,
++    recovery_available: true,
++    static_ndk_lib: true,
++    stl: "none",
++    local_include_dirs: ["upstream-freebsd/android/include/",],
++    cflags: [
++        "-D__BIONIC_LP32_USE_LONG_DOUBLE",
++        "-D_BSD_SOURCE",
++        "-DFLT_EVAL_METHOD=0",
++        "-include freebsd-compat.h",
++        "-fno-math-errno",
++        "-Wall",
++        "-Werror",
++        "-Wno-ignored-pragmas",
++        "-Wno-missing-braces",
++        "-Wno-parentheses",
++        "-Wno-sign-compare",
++        "-Wno-unknown-pragmas",
++        "-Wno-unused-const-variable",
++        "-Wno-unused-variable",
++        "-Wno-unused-parameter",
++        "-O2",
++        "-ffp-contract=fast",
++        "-DWANT_ERRNO_UFLOW",
++        "-mfma",
++        "-mavx2",
++    ],
++    arch: {
++        x86: {
++                srcs : [
++                        "x86/kabylake/acos_avx2.c",
++                        "x86/kabylake/cosf_avx2.c",
++                        "x86/kabylake/log_avx2.c",
++                        "x86/kabylake/logf_avx2.c",
++                        "x86/kabylake/sincosf_avx2.c",
++                        "x86/kabylake/sinf_avx2.c",
++                        "x86/kabylake/sinh_avx2.c",
++                        "x86/kabylake/tanh_avx2.c",
++		],
++        },
++
++        x86_64: {
++		srcs : [
++                        "x86_64/kabylake/acos_avx2.c",
++                        "x86_64/kabylake/cosf_avx2.c",
++                        "x86_64/kabylake/expm1_avx2.c",
++                        "x86_64/kabylake/log_avx2.c",
++                        "x86_64/kabylake/logf_avx2.c",
++                        "x86_64/kabylake/sincosf_avx2.c",
++                        "x86_64/kabylake/sinf_avx2.c",
++                        "x86_64/kabylake/sinh_avx2.c",
++                        "x86_64/kabylake/tanh_avx2.c",
++		],
++        },
++    },
++    
++    include_dirs: ["bionic/libc"],
++    system_shared_libs: ["libc"],
++    apex_available: [
++           "//apex_available:platform",
++           "com.android.runtime",
++    ],
++}
+diff --git a/libm/upstream-freebsd/lib/msun/src/e_acos.c b/libm/upstream-freebsd/lib/msun/src/e_acos.c
+index 1f6dca5bb..db5a6d81a 100644
+--- a/libm/upstream-freebsd/lib/msun/src/e_acos.c
++++ b/libm/upstream-freebsd/lib/msun/src/e_acos.c
+@@ -105,7 +105,8 @@ __ieee754_acos(double x)
+ 	    return 2.0*(df+w);
+ 	}
+ }
+-
++/*
+ #if LDBL_MANT_DIG == 53
```

Comment:
This #if is missing wen we moved the weak references to dynamic dispatch

Reply:
added this #if in dynamic dispatch